core/components/tree/basenode.js

```diff
@@ -0,0 +1,1151 @@
+};
+
+/**
+ * @return {Element} The div containing the children.
+ * @protected
+ */
+Blockly.tree.BaseNode.prototype.getChildrenElement = function() {
+  var el = this.getElement();
+  return el ? /** @type {Element} */ (el.lastChild) : null;
+};
+
+/**
+ * Sets the icon class for the node.
+ * @param {string} s The icon class.
+ */
+Blockly.tree.BaseNode.prototype.setIconClass = function(s) {
+  this.iconClass_ = s;
+  if (this.isInDocument()) {
+    this.updateIcon_();
+  }
+};
+
+/**
+ * Gets the icon class for the node.
+ * @return {string} s The icon source.
+ */
+Blockly.tree.BaseNode.prototype.getIconClass = function() {
+  return this.iconClass_;
+};
+
+/**
+ * Sets the icon class for when the node is expanded.
+ * @param {string} s The expanded icon class.
+ */
+Blockly.tree.BaseNode.prototype.setExpandedIconClass = function(s) {
+  this.expandedIconClass_ = s;
+  if (this.isInDocument()) {
+    this.updateIcon_();
+  }
+};
+
+/**
+ * Gets the icon class for when the node is expanded.
+ * @return {string} The class.
+ */
+Blockly.tree.BaseNode.prototype.getExpandedIconClass = function() {
+  return this.expandedIconClass_;
+};
+
+/**
+ * Sets the text of the label.
+ * @param {string} s The plain text of the label.
+ */
+Blockly.tree.BaseNode.prototype.setText = function(s) {
+  this.content_ = s;
+};
+
+/**
+ * Returns the text of the label. If the text was originally set as HTML, the
+ * return value is unspecified.
+ * @return {string} The plain text of the label.
+ */
+Blockly.tree.BaseNode.prototype.getText = function() {
+  return this.content_;
+};
+
+/**
+ * Sets the text of the tooltip.
+ * @param {string} s The tooltip text to set.
+ */
+Blockly.tree.BaseNode.prototype.setToolTip = function(s) {
+  this.toolTip_ = s;
+  var el = this.getLabelElement();
+  if (el) {
+    el.title = s;
+  }
+};
+
+/**
+ * Returns the text of the tooltip.
+ * @return {?string} The tooltip text.
+ */
+Blockly.tree.BaseNode.prototype.getToolTip = function() {
+  return this.toolTip_;
+};
+
+/**
+ * Updates the row styles.
+ */
+Blockly.tree.BaseNode.prototype.updateRow = function() {
+  var rowEl = this.getRowElement();
+  if (rowEl) {
+    rowEl.className = this.getRowClassName();
+  }
+};
+
+/**
+ * Updates the expand icon of the node.
+ */
+Blockly.tree.BaseNode.prototype.updateExpandIcon = function() {
+  var img = this.getExpandIconElement();
+  if (img) {
+    img.className = this.getExpandIconClass();
+  }
+  var cel = this.getChildrenElement();
+  if (cel) {
+    cel.style.backgroundPosition = this.getBackgroundPosition();
+  }
+};
+
+/**
+ * Updates the icon of the node. Assumes that this.getElement() is created.
+ * @private
+ */
+Blockly.tree.BaseNode.prototype.updateIcon_ = function() {
+  this.getIconElement().className = this.getCalculatedIconClass();
+};
+
+/**
+ * Handles mouse down event.
+ * @param {!Event} e The browser event.
+ * @protected
+ */
+Blockly.tree.BaseNode.prototype.onMouseDown = function(e) {
```

Comment:
It looks like if we want to we'll be able to wire this into gesture, instead of having click handling be completely separate for the toolbox.  Could be interesting to explore.

Reply:
And I see below that you're using `bindEventWithChecks`.  Cool.